src/EA.Weee.Database/scripts/Update/Release 7/Sprint7/20241105-1120-producer-eee-sp-update-returned.sql

```diff
@@ -0,0 +1,204 @@
+    SELECT 
+        RP.ProducerRegistrationNumber AS 'PRN',
+        RP.SchemeId,
+        S.ApprovalNumber,
+        S.SchemeName,
+        COALESCE(PBC.Name, PBP.Name, '') AS 'ProducerName',
+        COALESCE(ROC_A_C.Name, PPOB_A_C.Name, '') AS 'ProducerCountry',
+        DR.Quarter,
+        EEOA.WeeeCategory,
+        EEOA.Tonnage,
+        0 AS IsDirectProducer  -- Not a direct producer
+    FROM [PCS].DataReturn DR
+    INNER JOIN [PCS].DataReturnVersion DRV ON DR.CurrentDataReturnVersionId = DRV.Id
+    INNER JOIN [PCS].[EeeOutputReturnVersion] EEORV ON DRV.EeeOutputReturnVersionId = EEORV.Id
+    INNER JOIN [PCS].[EeeOutputReturnVersionAmount] EEORVA ON EEORV.Id = EEORVA.EeeOutputReturnVersionId
+    INNER JOIN [PCS].[EeeOutputAmount] EEOA ON EEORVA.EeeOuputAmountId = EEOA.Id
+    INNER JOIN [Producer].[RegisteredProducer] RP ON EEOA.RegisteredProducerId = RP.Id
+    INNER JOIN [PCS].Scheme S ON RP.SchemeId = S.Id
+    INNER JOIN [Producer].[ProducerSubmission] PS ON RP.[CurrentSubmissionId] = PS.[Id]
+    LEFT JOIN [Producer].[Business] PB ON PS.ProducerBusinessId = PB.Id
+    LEFT JOIN [Producer].[Company] PBC ON PB.CompanyId = PBC.Id
+    LEFT JOIN [Producer].[Partnership] PBP ON PB.PartnershipId = PBP.Id
+    LEFT JOIN [Producer].[Contact] ROC 
+        INNER JOIN [Producer].[Address] ROC_A 
+            INNER JOIN [Lookup].[Country] ROC_A_C ON ROC_A.CountryId = ROC_A_C.Id     
+        ON ROC.AddressId = ROC_A.Id
+    ON PBC.RegisteredOfficeContactId = ROC.Id
+    LEFT JOIN [Producer].[Contact] PPOB 
+        INNER JOIN [Producer].[Address] PPOB_A 
+            INNER JOIN [Lookup].[Country] PPOB_A_C ON PPOB_A.CountryId = PPOB_A_C.Id   
+        ON PPOB.AddressId = PPOB_A.Id
+    ON PBP.PrincipalPlaceOfBusinessId = PPOB.Id
+    WHERE DR.ComplianceYear = @ComplianceYear
+        AND RP.Removed = 0
+        AND EEOA.ObligationType = @ObligationType
+        AND (@SchemeId IS NULL OR S.[Id] = @SchemeId)
+        AND (@FilterByDirectRegistrant = 0)
+
+    UNION ALL
+
+    SELECT
+        RP.ProducerRegistrationNumber as 'PRN',
+        dr.Id AS 'SchemeId',
+        '' AS 'ApprovalNumber',
+        'Direct registrant' AS SchemeName,
+        COALESCE(ap.OverseasProducerName, o.[Name], '') AS 'ProducerName',
+        COALESCE(ac.Name, oc.Name, '') AS 'ProducerCountry',
+        4 AS Quarter,
+        eoa.WeeeCategory,
+        eoa.Tonnage,
+        1 AS IsDirectProducer  -- This is a direct producer
+    FROM
+        [Producer].[DirectProducerSubmission] DPS
+		INNER JOIN [Producer].[DirectRegistrant] DR ON DR.Id = DPS.DirectRegistrantId
+		INNER JOIN [Organisation].[Organisation] O ON O.Id = DR.OrganisationId
+		INNER JOIN [Organisation].[Address] OA ON OA.Id = O.BusinessAddressId
+		INNER JOIN [Lookup].[Country] OC ON OC.Id = OA.CountryId
+		INNER JOIN [Producer].[RegisteredProducer] RP ON DPS.RegisteredProducerId = RP.Id
+		INNER JOIN (
+			SELECT 
+				DirectProducerSubmissionId,
+				EeeOutputReturnVersionId,
```

Comment:
Change this to join on most recently submitted rather than the current submission.